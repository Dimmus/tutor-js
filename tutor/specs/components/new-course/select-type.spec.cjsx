{React, spyOnComponentMethod, SnapShot} = require '../helpers/component-testing'

SelectType = require '../../../src/components/new-course/select-type'
{NewCourseStore} = require '../../../src/flux/new-course'

describe 'CreateCourse: Selecting type of course', ->

  it 'it sets state when type is clicked', ->
    spy = spyOnComponentMethod(SelectType, 'onSelectType')
    wrapper = mount(<SelectType />)
    expect(NewCourseStore.get('course_type')).to.exist
    wrapper.find('[data-brand="coach"]').simulate('click')
    expect(spy.calledOnce).to.be.true
    undefined

  it 'matches snapshot', ->
    component = SnapShot.create(
      <SelectType />
    )
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
