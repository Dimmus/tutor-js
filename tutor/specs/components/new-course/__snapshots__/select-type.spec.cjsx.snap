exports[`CreateCourse: Selecting type of course matches snapshot 1`] = `
<div
  className="list-group">
  <div
    className="list-group-item choice"
    data-brand="tutor-beta"
    onClick={[Function]} />
  <div
    className="list-group-item choice active"
    data-brand="coach"
    onClick={[Function]} />
</div>
`;
